tests/vmi_configuration_test.go

```diff
@@ -346,6 +347,88 @@ var _ = Describe("Configurations", func() {
 				Expect(err).ToNot(HaveOccurred())
 			})
 		})
+
+		Context("with guestAgent", func() {
+			var agentVMI *v1.VirtualMachineInstance
+
+			It("there should be agent channel by default", func() {
+
+				agentVMI = tests.NewRandomVMIWithEphemeralDisk(tests.RegistryDiskFor(tests.RegistryDiskAlpine))
+				By("Starting a VirtualMachineInstance")
+				agentVMI, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(agentVMI)
+				Expect(err).ToNot(HaveOccurred(), "Should create VMI successfully")
+				tests.WaitForSuccessfulVMIStart(agentVMI)
+
+				getOptions := metav1.GetOptions{}
+				var freshVMI *v1.VirtualMachineInstance
+
+				freshVMI, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(agentVMI.Name, &getOptions)
+				Expect(err).ToNot(HaveOccurred(), "Should get VMI ")
+
+				domXML := getVmDomainXml(virtClient, freshVMI)
+
+				Expect(domXML).To(ContainSubstring("<channel type='unix'>"), "Should contain at least one channel")
+				Expect(domXML).To(ContainSubstring("<target type='virtio' name='org.qemu.guest_agent.0' state='disconnected'/>"), "Should have guest agent channel present")
+				Expect(domXML).To(ContainSubstring("<alias name='channel0'/>"), "Should have guest channel present")
+			})
+
+			It("should report on the domain xml that the guest is connected", func() {
+				vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(tests.RegistryDiskFor(tests.RegistryDiskFedora), fmt.Sprintf(`#!/bin/bash
+					echo "fedora" |passwd fedora --stdin
+					mkdir -p /usr/local/bin
+					curl %s > /usr/local/bin/qemu-ga
+					chmod +x /usr/local/bin/qemu-ga
+					setenforce 0
+					systemd-run /usr/local/bin/qemu-ga
+					`, tests.GuestAgentHttpUrl))
+				vmi.Spec.Domain.Resources.Requests[kubev1.ResourceMemory] = resource.MustParse("512M")
+
+				By("Starting a VirtualMachineInstance")
+				vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+				Expect(err).ToNot(HaveOccurred())
+				tests.WaitForSuccessfulVMIStart(vmi)
+
+				Eventually(func() string {
+					domXml, err := tests.GetRunningVirtualMachineInstanceDomainXML(virtClient, vmi)
+					Expect(err).ToNot(HaveOccurred())
+					return domXml
+				}, 180*time.Second, 2*time.Second).Should(ContainSubstring(`<target type='virtio' name='org.qemu.guest_agent.0' state='connected'/>`))
+			})
+
+			It("VMI condition shloud signal agent presence", func() {
+
+				// TODO: actually review this once the VM image is present
+				agentVMI := tests.NewRandomVMIWithEphemeralDiskAndUserdata(tests.RegistryDiskFor(tests.RegistryDiskFedora), fmt.Sprintf(`#!/bin/bash
+                echo "fedora" |passwd fedora --stdin
```

Comment:
Can you deduplicate the cloud-init content?

Reply:
Yep, already done, I removed one test. It turned out it is testing exactly the same thing. So no need for two of them.